src/useCases/SendNotificationUseCase.ts

```diff
@@ -9,24 +10,19 @@ import {
   NewNotificationRecord,
   NewNotificationRepository,
 } from '../domain/NewNotificationRepository';
-import { Logger } from '../logger';
-import { onValidApiKey } from './utils';
+import { authorizeApiKey } from './utils';
 
 export const SendNotificationUseCase =
-  (logger: Logger, repository: NewNotificationRepository) =>
+  (repository: NewNotificationRepository) =>
   (apiKey: ApiKey) =>
-  (body: NewNotificationRequest): TE.TaskEither<Error, NewNotificationRecord['output']> => {
-    const input = { apiKey, body };
-    const returned = makeNewNotificationResponse(body)(crypto.randomUUID());
-    const output = onValidApiKey(apiKey)({ statusCode: 202 as const, returned });
-    const record = makeNewNotificationRecord({ input, output });
-    return pipe(
-      repository.insert(record),
-      TE.map((_) => {
-        logger.debug(record);
-        return record.output;
-      })
+  (body: NewNotificationRequest): TE.TaskEither<Error, NewNotificationRecord['output']> =>
+    pipe(
+      // authorize the key
+      authorizeApiKey(apiKey),
+      // create the response on valid key
+      E.map((_) => ({ statusCode: 202 as const, returned: makeNewNotificationResponse(body)(crypto.randomUUID()) })),
+      E.toUnion,
+      (output) => repository.insert(makeNewNotificationRecord({ input: { apiKey, body }, output })),
```

Comment:
In my opinion this part is more readable in this form:

```suggestion
      // create the response on valid key
      E.map((_) => {
        const returned = makeNewNotificationResponse(body)(crypto.randomUUID());
        const output = { statusCode: 202 as const, returned };
        return makeNewNotificationRecord({ input: { apiKey, body }, output });
      }),
      E.toUnion,
      repository.insert,
```
or in this other one:

```suggestion
// create the response on valid key
      E.map((_) => makeNewNotificationResponse(body)(crypto.randomUUID())),
      E.map((returned) => ({ statusCode: 202 as const, returned })),
      E.map((output) => makeNewNotificationRecord({ input: { apiKey, body }, output })),
      E.toUnion,
      repository.insert,
```

Reply:
I agree with you and I think the second option is better to read